File: ruby/examples/quickstart.rb
# frozen_string_literal: true

$LOAD_PATH.unshift File.expand_path("../lib", __dir__)
require "gmaps_scraper"

client = GmapsScraper::Client.new
me = client.me
puts "plan=#{me['plan']} creditsRemaining=#{me['creditsRemaining']}"

rows = client.scrape("dentists in Austin TX")
puts "scraped #{rows.length} places"
unless rows.empty?
  r = rows.first
  puts "#{r['Name']} #{r['Phone']} #{r['Website']} #{r['Emails']}"
end

File: rust/examples/quickstart.rs
use google_maps_scraper_sdk::{Client, ClientOptions};

fn main() {
    let client = Client::new(ClientOptions::default()).expect("client");
    let me = client.me().expect("me");
    println!(
        "plan={} creditsRemaining={}",
        me.plan, me.credits_remaining
    );

    let rows = client
        .scrape("dentists in Austin TX", Default::default())
        .expect("scrape");
    println!("scraped {} places", rows.len());
    if let Some(row) = rows.first() {
        println!(
            "{:?} {:?} {:?} {:?}",
            row.get("Name"),
            row.get("Phone"),
            row.get("Website"),
            row.get("Emails")
        );
    }
}
